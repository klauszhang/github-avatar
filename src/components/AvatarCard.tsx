import React from 'react'
import { Box, Card, CardMedia, CardContent } from '@mui/material'
import EllipsisContentTypographyProps from './EllipsisContentTypography'

interface AvatarCardProps {
  cardName: string
  imgUrl: string
  imgBgColor: string // The hex color code, without the leading '#'
  description: string
}

const AvatarCard: React.FC<AvatarCardProps> = ({
  cardName,
  imgUrl,
  imgBgColor,
  description,
}) => {
  const imgBgColorWithSharp = '#' + imgBgColor

  return (
    <Card
      raised={true}
      sx={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        maxWidth: 400,
        maxHeight: 600, 
        minWidth: 300, 
        minHeight: 400, 
        borderRadius: 5,
      }}
    >
      <Box
        sx={{
          display: 'flex',
          justifyContent: 'center',
          width: '100%',
          p: 2,
          backgroundColor: imgBgColorWithSharp,
        }}
      >
        <CardMedia
          component="img"
          sx={{ maxWidth: 200 }}
          image={imgUrl}
          alt={cardName}
        />
      </Box>
      <CardContent sx={{ p: 3, width: '100%' }}>
        <Box>
          <EllipsisContentTypographyProps variant="h6" sx={{ mb: 1 }} lines={2}>
            {cardName}
          </EllipsisContentTypographyProps>
          <EllipsisContentTypographyProps variant="body1" lines={3}>
            {description}
          </EllipsisContentTypographyProps>
        </Box>
      </CardContent>
    </Card>
  )
}

export default AvatarCard
