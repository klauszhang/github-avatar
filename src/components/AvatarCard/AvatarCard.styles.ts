export const CardContainer ={
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    maxWidth: 400,
    borderRadius: 5,
}

export const Box = {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
    p: 2,
}

export const CardContent = {
    p: 3, 
    width: '100%'
}

export const Typography = { mb: 1, textAlign: 'center' }

export const CardMedia = { maxWidth: 200 }