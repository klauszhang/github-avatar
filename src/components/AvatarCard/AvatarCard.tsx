import React from 'react'
import { Box, Card, CardMedia, CardContent, Link } from '@mui/material'
import EllipsisContentTypographyProps from '../EllipsisContentTypography'
import * as styles from './AvatarCard.styles'

interface AvatarCardProps {
  cardName: string
  imgUrl: string
  url: string 
}

const AvatarCard: React.FC<AvatarCardProps> = ({
  cardName,
  imgUrl,
  url,
}) => {
  return (
    <Card
      raised={true}
      sx={styles.CardContainer}
    >
      <Box
        sx={styles.Box}
      >
        <CardMedia
          component="img"
          sx={styles.CardMedia}
          image={imgUrl}
          alt={cardName}
        />
      </Box>
      <CardContent sx={styles.CardContent}>
        <Box>
          
          <EllipsisContentTypographyProps variant="h6" lines={2} sx={styles.Typography}>
            <Link href={url} underline="hover" target="_blank" rel="noreferrer">
              {cardName}
            </Link> 
          </EllipsisContentTypographyProps>
        </Box>
      </CardContent>
    </Card>
  )
}

export default AvatarCard
