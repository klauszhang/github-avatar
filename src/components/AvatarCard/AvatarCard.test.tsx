import { render, screen } from '@testing-library/react'
import AvatarCard from './AvatarCard'

test('renders AvatarCard', () => {
  const {container} = render(<AvatarCard  cardName='card name' imgUrl='imgUrl' url='url' />)
  const cardName = screen.getByText('card name')
  expect(cardName).toBeInTheDocument()
  expect(container).toMatchSnapshot()
  expect(cardName.closest('a')).toHaveAttribute('href', 'url')
})
