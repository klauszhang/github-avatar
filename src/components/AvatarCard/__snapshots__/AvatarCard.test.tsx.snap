// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AvatarCard 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-1q1eqzk-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-7h59a4"
    >
      <img
        alt="card name"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-4wvnlr-MuiCardMedia-root"
        src="imgUrl"
      />
    </div>
    <div
      class="MuiCardContent-root css-1fp245t-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1tpu4y-MuiTypography-root"
          lines="2"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17sj1r4-MuiTypography-root-MuiLink-root"
            href="url"
            rel="noreferrer"
            target="_blank"
          >
            card name
          </a>
        </h6>
      </div>
    </div>
  </div>
</div>
`;
